### PR TITLE
Allow using items on any monster in your party during combat. Closes #794

### LINKED ISSUE #794: Revive item not usable in combat
Revive requires that the target be unconscious, but using an item in combat requires that you target a monster that is in the combat.

Ideally we'd be prompted to choose from our current team, as we do when using the item out of combat.

### PR DESCRIPTION
**Pull request by ultidonki**
This PR allows you to use items on any monster in your party, during a battle. 

It adds a new optional property to Items - battle_menu. 

If battle_menu is == "monster", then when you use this item during battle, the Monster menu will appear, and you can select any monster in your party to use the item on. 
If battle_menu is == "combat", or not set, then you can choose between your current monster and the enemy monster, as normal. 

I've also added battle_menu to the db.py validation and schemas/item-schema.json, which seems to work (it gets rid of the error messages) but it's my first time updating them so it could be wrong. 

Hopefully adding the battle_menu property was a good idea, but if anyone has any better suggestions I'm happy to change it to something else! 

This should close issue #794

### CODE REVIEW

File: tuxemon/item/item.py [hunk 1]
```diff
@@ -154,6 +155,7 @@ def load(self, slug: str) -> None:
         self.usable_in = results["usable_in"]
         self.target = process_targets(results["target"])
         self.effects = self.parse_effects(results.get("effects", []))
+        self.battle_menu = (results["battle_menu"] if "battle_menu" in results else "")
```
Comment (kerizane) [inline tuxemon/item/item.py]: self.battle_menu = results.get("battle_menu", "")

**Review by kerizane — DISMISSED** [review]

File: tuxemon/item/item.py (hunk 1 above)
Comment (ultidonki) [reply tuxemon/item/item.py]: > self.battle_menu = results.get("battle_menu", "")

Thankyou, I've made this change now. Thanks!

File: tuxemon/db.py [hunk 2]
```diff
@@ -35,15 +35,7 @@
...
-from typing import (
-    Any,
-    Dict,
-    Literal,
-    Mapping,
-    Optional,
-    Sequence,
-    overload,
-)
+from typing import Any, Dict, Literal, Mapping, Optional, Sequence, overload
```
Comment (kerizane) [inline tuxemon/db.py:38]: Picky, but should this have changed? Could you run isort and black against this file? (Just run script/fmt.sh and only commit the change to files that are currently in this PR).

If this causes lots of unrelated changes then let me know.

Comment (kerizane) [reply tuxemon/db.py:38]: Looks good other than that.

**Review by kerizane — CHANGES_REQUESTED** [review]

**Review by kerizane — APPROVED** [review]

**Comment by kerizane** [comment]
Thanks for the fix.

**Comment by ultidonki** [comment]
Sorry, I only had time to run fmt.sh and isort now, as you requested, but it looks like the PR was fine if it got accepted. Thanks for merging! 

I think the change it made (turning a one-line import into multiple lines), there was talking of possibly changing fmt.sh to stop doing that. Or maybe someone decided it's better on a multi-line, I can't remember, sorry. :smile:
